pyneuroml/utils/plot.py

```diff
@@ -389,3 +390,184 @@ def load_minimal_morphplottable__model(
                 for acell in inc.cells:
                     acell.biophysical_properties = None
                     nml_model.add(acell)
+
+
+def extract_coordinates_and_diameter(point_with_diam):
+    """
+    Extracts coordinates and diameter from a Point3DWithDiam object.
+
+    Parameters:
+        point_with_diam (neuroml.nml.nml.Point3DWithDiam): Object representing a point with diameter.
+
+    Returns:
+        tuple: Tuple containing coordinates and diameter.
+    """
+    coords = (point_with_diam.x, point_with_diam.y, point_with_diam.z)
+    diam = point_with_diam.diameter
+    return coords, diam
+
+
+def save_meshdata_to_file(meshdata, filename, format="obj", overwrite=True):
+    """Save the generated meshdata to a file.
+
+    Parameters
+    ----------
+    meshdata : dict
+        Dictionary containing meshdata.
+    filename : str
+        Filename to save the meshdata.
+    format : str
+        File format for the saved file. Supported formats are ".obj" and ".gz"
+    overwrite : bool
+        Boolean value specifying if file already exists and needs to be overwritten
+    """
+    if not meshdata:
+        logger.warning("No meshdata to save.")
+        return
+
+    logger.info(f"Saving meshdata to file: {filename}")
+
+    # Create directories if they don't exist
+    os.makedirs(os.path.dirname(filename), exist_ok=True)
+
+    vertices = []
+    faces = []
+    normals = []
+    texcoords = []
+
+    for key, segments in meshdata.items():
```

Comment:
So, we already create meshes in `create_instanced_meshes`, line 988, but we don't track them. Can we not just store these in a variable and then get the vertices etc. from this Meshdata instead of you basically recreating the mesh here which will be quite expensive for larger cells? 

See: https://vispy.org/api/vispy.geometry.meshdata.html#module-vispy.geometry.meshdata

From the test, it also appears that all the arguments are not necessary to generate the mesh:

https://github.com/vispy/vispy/blob/5d61ecbd930167446be0f9e5e078288e19839af9/vispy/io/tests/test_io.py#L67

Did you generate a mesh and see that it works in other tools? (Or did you just read it back in using vispy to see if it works?)

Reply:
Are you suggesting to use a ```mesh object``` instead of ```meshdata``` to store the mesh?
I did not explicitly generate a custom mesh to test this. I read a 3d mesh using vispy and tested it.

Reply:
So, if you look at our plotting code, we create cylindrical and spherical meshes ourselves. I am wondering if we can't just use that rather than you re-creating meshes. All of these meshes are stored as Meshdata---see `create_cylindrical_mesh` and the `meshdata` variable, which is passed to the different functions and holds all the meshdata we create